src/activities/multiplication-chart/multiplication-chart.js

```diff
@@ -0,0 +1,129 @@
+/* GCompris - multiplication-chart.js
+ *
+ * Copyright (C) 2016 Varun Kumar <varun13169@iiitd.ac.in>
+ *
+ * Authors:
+ *
+ *   Varun Kumar <varun13169@iiitd.ac.in>
+ *
+ *   This program is free software; you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation; either version 3 of the License, or
+ *   (at your option) any later version.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program; if not, see <http://www.gnu.org/licenses/>.
+ */
+.pragma library
+.import QtQuick 2.0 as Quick
+
+var currentLevel = 0
+var numberOfLevel = 9
+var items
+
+
+function start(items_) {
+    items = items_
+    currentLevel = 0
+    initLevel()
+}
+
+function stop() {
+}
+
+function initLevel() {
+    items.bar.level = currentLevel + 1
+    items.rowSelected = 0
+    items.colSelected = 0
+    items.enteredInput = 0
+    items.answer.text = qsTr("")
+
+    items.rowQues = Math.floor((Math.random() * 10) + 1);
+    items.colQues = Math.floor((Math.random() * 10) + 1);
+
+    for(var i=0; i<items.repeaterGridRow.count; i++) {        // For initialization of top rows color
+        items.repeaterGridRow.itemAt(i).color = "white"
+    }
+    for(i=0; i<items.repeaterGridCol.count; i++) {        // For initialization of side Column color
+        items.repeaterGridCol.itemAt(i).color = "white"
+    }
+    for(i=0; i<items.repeater.count; i++) {        // For initialization of side Column color
+        items.repeater.itemAt(i).color = "green"
+    }
+}
+
+function nextLevel() {
+    if(numberOfLevel <= ++currentLevel ) {
+        currentLevel = 0
+    }
+    initLevel();
+}
+
+function previousLevel() {
+    if(--currentLevel < 0) {
+        currentLevel = numberOfLevel - 1
+    }
+
+    initLevel();
+}
+
+
+
+function makeOtherColInRowWhite() {
+    for(var i=0; i<items.repeaterGridRow.count; i++) {
```

Comment:
Can't the state be directly binded to the value of the selected row/column? I mean, in the Repeater having something like:
state: index == items.rowSelected ? "active" : "default"